rules/Naming/ExpectedNameResolver/InflectorSingularResolver.php

```diff
@@ -17,6 +17,7 @@ final class InflectorSingularResolver
      */
     private const SINGULARIZE_MAP = [
         'news' => 'new',
+        'parametersAcceptor' => 'parametersAcceptor',
```

Comment:
how about check if object exists?, it can be also `ParametersAcceptorWithPhpDoc`

Reply:
we don't know the namespace of the class

Reply:
On `RenameForeachValueVariableToMatchExprVariableRector` usage, the `$stmt->valueVar` or `$stmt->expr` may need to be checked by its type, if it got `ObjectType`, then only make snake case without singularize.

Reply:
checking `s` in the middle seems make `psr4NamespacesToPaths` not changed, so checking object type instead on `RenameForeachValueVariableToMatchExprVariableRector` is probably the way to go

Reply:
we have tests from the past which explicitly want `s` beeing stripped: https://github.com/rectorphp/rector-src/pull/5200/files#diff-3f99d6810b94a42bb1c18142c9ef4d37718da7d41bbaf96accd7e58ace82f3e8R39